flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -480,13 +483,17 @@ private StreamGraph generateInternal(List<StreamTransformation<?>> transformatio
 		streamGraph.addSource(source.getId(),
 				slotSharingGroup,
 				source.getCoLocationGroupKey(),
-				source.getOperator(),
+				source.getOperatorFactory(),
 				null,
 				source.getOutputType(),
 				"Source: " + source.getName());
-		if (source.getOperator().getUserFunction() instanceof InputFormatSourceFunction) {
-			InputFormatSourceFunction<T> fs = (InputFormatSourceFunction<T>) source.getOperator().getUserFunction();
-			streamGraph.setInputFormat(source.getId(), fs.getFormat());
+		if (source.getOperatorFactory() instanceof SimpleOperatorFactory) {
+			SimpleOperatorFactory<T> factory = (SimpleOperatorFactory<T>) source.getOperatorFactory();
+			Function userFunction = ((StreamSource) factory.getOperator()).getUserFunction();
+			if (userFunction instanceof InputFormatSourceFunction) {
+				InputFormatSourceFunction<T> fs = (InputFormatSourceFunction<T>) userFunction;
+				streamGraph.setInputFormat(source.getId(), fs.getFormat());
```

Comment:
Here need get input format, not exposing this capability to StreamOperatorFactory